src/System.Buffers.ReaderWriter/System/Buffers/Reader/BufferReader_text.cs

```diff
@@ -3,138 +3,700 @@
+        /// <param name="value">The parsed <see cref="ushort"/> value or default if failed.</param>
+        /// <param name="standardFormat">Expected format.</param>
+        /// <remarks>
+        /// <see cref="Utf8Parser.TryParse(ReadOnlySpan{byte}, out ushort, out int, char)"/> for supported formats.
+        /// </remarks>
+        /// <exception cref="FormatException">Thrown if the given format isn't supported.</exception>
+        /// <returns>Number of consumed bytes if successful, otherwise 0.</returns>
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public static unsafe bool TryParse(ref this BufferReader<byte> reader, out int value)
+        public static unsafe int TryParse(
+            ref this BufferReader<byte> reader,
+            out ushort value,
+            char standardFormat = '\0')
         {
             ReadOnlySpan<byte> unread = reader.UnreadSpan;
-            if (Utf8Parser.TryParse(unread, out value, out int consumed) && consumed < unread.Length)
+
+            if (!Utf8Parser.TryParse(unread, out value, out int consumed, standardFormat)
+                || consumed >= MaxParseBytes)
+            {
+                // Failed to parse or we consumed more than we're willing to accept
+                return 0;
+            }
+            else if (consumed < unread.Length)
             {
+                // The parser found a value it wouldn't consume so we have all useful data.
                 reader.Advance(consumed);
-                return true;
+                return consumed;
             }
 
-            return TryParseSlow(ref reader, out value);
+            // If we ate all of our unread there may be more valid digits.
+            // The typical max size for a short will be less than 7 bytes ("65,535").
+            return TryParseSlow(ref reader, out value, standardFormat, 7, UshortParser);
         }
 
-        private static unsafe bool TryParseSlow(ref BufferReader<byte> reader, out int value)
+        /// <summary>
+        /// Try to parse an <see cref="int"/> out of the reader (as UTF-8).
+        /// </summary>
+        /// <param name="value">The parsed <see cref="int"/> value or default if failed.</param>
+        /// <param name="standardFormat">Expected format.</param>
+        /// <remarks>
+        /// <see cref="Utf8Parser.TryParse(ReadOnlySpan{byte}, out int, out int, char)"/> for supported formats.
+        /// </remarks>
+        /// <exception cref="FormatException">Thrown if the given format isn't supported.</exception>
+        /// <returns>Number of consumed bytes if successful, otherwise 0.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static unsafe int TryParse(
+            ref this BufferReader<byte> reader,
+            out int value,
+            char standardFormat = '\0')
         {
-            const int MaxLength = 15;
+            //return TryParseFast(ref reader, out value, standardFormat, 15, IntParser);
```

Comment:
remove commented out code

Reply:
Whoops- remnant of discovering there is no way to do generics here without significant overhead. :(